File: JavaScript Language/JavaScript Language/Inheritance.js
    // javascript - Inheritance
    function println(params){
        console.log(params );
    }
    println("Hello World");

    // inheritance is a feature by which one cllass acquire properties of another class
    // extends keyword is used for inheriting properties
    // example
    class Vehicle{
        constructor(){
            this.Company = "Honda";
        }
    }
    class Car extends Vehicle{
        constructor(Price,Color){
            super(); // super keyword is used to call parent constructor
            this.Price = 10;
            this.Color = "red";
        }
        DisplayData(){
            println("Company : "+this.Company);
            println("Price : "+this.Price);
            println("Color : "+this.Color);
        }
    }
    c1 = new Car(10,"red");
    c1.DisplayData();

    // example 2
    class Person{
        constructor(Height,Weight,Age){
            this.Height = Height;
            this.Weight = Weight;
            this.Age = Age;
        }
    }
    class Student extends Person{
        constructor(Height,Weight,Age){
            super(Height,Weight,Age);
        }
        DisplayData(){
            println("Age = "+this.Age);
            println("Height = "+this.Height);
            println("Weight = "+this.Weight);
        }
    }

    s1 = new Student(180,50,18);
    s1.DisplayData();

File: Java/Java OOP/Inheritance.java
// java - Inheritance
public class Inheritance {
    public static void main(String[] args) {
        System.out.println("Hello World");
        Car c = new Car();
        c.SetData("MYSelf", "Black", 10);
        c.DisplayData();
    }
}

// parent class
class Vehicle {
    String Owner;
    String color;
    float Price;

    void SetData(String Owner, String color, float Price) {
        this.Owner = Owner;
        this.color = color;
        this.Price = Price;
    }

    void DisplayData() {
        System.out.println("Vehicle Owner : " + this.Owner);
        System.out.println("Vehicle Detail : \ncolor " + this.color + "\nprice " + this.Price);
    }
}

// child class
class Car extends Vehicle {
    // overrides parent class methode
    @Override
    void DisplayData() {
        System.out.println("Car Owner : " + this.Owner);
        System.out.println("Car Detail : \ncolor " + this.color + "\nprice " + this.Price);
    }
}